### PR TITLE
Add Epinio service create form

### PR DESCRIPTION
**Pull request by catherineluse**
Screen Shot 2021-11-11 at 11 56 40 PM

### CODE REVIEW

**Comment by catherineluse** [comment]
@richard-cox I know you had the service deletions working in the past, but now it looks like the delete action isn't available from the service list view. I looked at the service model to see if there was anything different about the remove functionality on there, but didn't see any difference, even though apps and namespaces have the delete button and services don't. Do you know how the delete button could be re-enabled?

File: components/form/LabeledInput.vue [hunk 1]
```diff
@@ -36,7 +36,7 @@ export default {
       type:    Boolean,
     },
 
-    minHeight: {
+    minInputHeight: {
```
Comment (richard-cox) [inline components/form/LabeledInput.vue:39]: I think this one can now stay as is, I changed the type to `Number` so it matches the underlying type

File: assets/translations/en-us.yaml [hunk 2]
```diff
@@ -6156,6 +6156,7 @@ epinio:
     name: Name
     create: Create a Namespace
   services:
+    pairs: Configure data that will be available to a container
```
Comment (richard-cox) [inline assets/translations/en-us.yaml:6159]: I think this is better as `Environment Variables` with a `<i v-tooltip="'Environment variables are available to bound applications'" class="icon icon-info" />`

File: products/epinio/list/namespaces.vue [hunk 3]
```diff
@@ -128,7 +128,7 @@ export default {
         <div slot="body">
           <LabeledInput
             v-model="value.name"
-            :min-height="'90px'"
+            :min-input-height="'90px'"
```
Comment (richard-cox) [inline products/epinio/list/namespaces.vue:131]: This would then revert as well

File: products/epinio/list/services.vue [hunk 4]
```diff
@@ -22,7 +22,39 @@ export default {
...
     },
+    rows: {
+      type:     Array,
+      required: true,
+    },
   },
+
+  computed: {
+
+    filterRows() {
```
Comment (richard-cox) [inline products/epinio/list/services.vue:33]: This isn't used, is it needed?

File: products/epinio/models/services.js [hunk 5]
```diff
@@ -8,16 +8,16 @@ import EpinioResource from './epinio-resource';
 export default class EpinioService extends EpinioResource {
   get links() {
     return {
-      update: this.getUrl(),
-      self:   this.getUrl(),
-      remove: this.getUrl(),
-      create: this.getUrl(this.meta?.namespace, null), // ensure name is null
+      update: this.getUrl(this.meta?.namespace, this.meta?.name),
```
Comment (richard-cox) [inline products/epinio/models/services.js:11]: update, self and remove won't need these params as they're the functions defaults

File: products/epinio/models/services.js [hunk 6]
```diff
@@ -8,16 +8,16 @@ import EpinioResource from './epinio-resource';
...
+      create: this.getUrl(this.meta?.namespace, ''),
     };
   }
 
-  getUrl(namespace = this.meta?.namespace, name = this.meta?.name) {
-    // Add baseUrl in a generic way
-    return this.$getters['urlFor'](this.type, this.id, { url: `/api/v1/namespaces/${ namespace }/services/${ name || '' }` });
+  getUrl(namespace = this.meta?.namespace) {
+    // Don't include the name when creating a service.
+    return this.$getters['urlFor'](this.type, this.id, { url: `/api/v1/namespaces/${ namespace }/services/${ name }` });
```
Comment (richard-cox) [inline products/epinio/models/services.js:20]: name is not defined

File: products/epinio/models/services.js [hunk 7]
```diff
@@ -56,5 +56,68 @@ export default class EpinioService extends EpinioResource {
     });
   }
 
+  get hasCustomList() {
```
Comment (richard-cox) [inline products/epinio/models/services.js:59]: Some of these i don't think are needed. Can you confirm why each were added?

**Review by richard-cox — CHANGES_REQUESTED** [review]
I'm gonna try to blitz these today, along with the delete service issue

File: assets/translations/en-us.yaml (hunk 2 above)
Comment (richard-cox) [reply assets/translations/en-us.yaml]: I got this wrong, it should still reference data

**Review by richard-cox — APPROVED** [review]

File: components/form/LabeledInput.vue (hunk 1 above)
Comment (catherineluse) [reply components/form/LabeledInput.vue]: It refers to a css property like "50px" so should be a string

File: products/epinio/list/namespaces.vue (hunk 3 above)
Comment (catherineluse) [reply products/epinio/list/namespaces.vue]: If it's min-height it conflicts with a property in another component

File: products/epinio/models/services.js (hunk 6 above)
Comment (catherineluse) [reply products/epinio/models/services.js]: The problem with using a default value for the service name was that it was using the service name when trying to create it. Using a default name gave it the incorrect route to post to.

**Pull request merged**